src/blocks/Content/Component.tsx

```diff
@@ -1,41 +1,49 @@
 import RichText from '@/components/RichText'
 import { cn } from '@/utilities/ui'
+import Color from 'color'
 
 import type { ContentBlock as ContentBlockProps } from '@/payload-types'
 
-import { CMSLink } from '@/components/Link'
-
 export const ContentBlock = (props: ContentBlockProps) => {
-  const { columns } = props
+  const { enableColumns, columns, content, color } = props
+  const numOfCols = columns?.length ?? 1
+
+  const bgColorClass = `bg-[${color}]`
 
-  const colsSpanClasses = {
-    full: '12',
-    half: '6',
-    oneThird: '4',
-    twoThirds: '8',
+  const colsClasses: { [key: number]: string } = {
+    1: 'lg:col-span-12',
+    2: 'lg:col-span-6',
+    3: 'lg:col-span-4',
+    4: 'lg:col-span-3',
   }
+  const colsSpanClass = colsClasses[numOfCols]
 
+  // TODO - import color list from theme
+  const bgColor = Color(`${color}`)
+  const textColor = bgColor.isLight() ? 'text-black' : 'text-white'
   return (
-    <div className="container my-16">
-      <div className="grid grid-cols-4 lg:grid-cols-12 gap-y-8 gap-x-16">
-        {columns &&
-          columns.length > 0 &&
-          columns.map((col, index) => {
-            const { enableLink, link, richText, size } = col
-
-            return (
-              <div
-                className={cn(`col-span-4 lg:col-span-${colsSpanClasses[size || 'full']}`, {
-                  'md:col-span-2': size !== 'full',
-                })}
-                key={index}
-              >
-                {richText && <RichText data={richText} enableGutter={false} />}
+    <div className={`${bgColorClass}`}>
+      <div className="container py-16">
+        <div className="grid grid-cols-4 lg:grid-cols-12 gap-y-8 gap-x-16">
+          {enableColumns ? (
+            columns?.map((col, index) => {
+              const { richText } = col
 
-                {enableLink && <CMSLink {...link} />}
-              </div>
-            )
-          })}
+              return (
+                <div
+                  className={cn(`col-span-4 md:col-span-2 ${colsSpanClass} ${textColor}`)}
```

Comment:
Def a bit of a preference thing in this scenario since colsSpanClass and textColor will never be undefined but I find this a little easier to write too. Up to you!

```suggestion
                  className={cn('col-span-4 md:col-span-2', colsSpanClass, textColor)}
```

Could update on line 42 as well if you like this style better.

Reply:
Makes sense to me to separate them by a comma.